bc/subscription/tasks.py

```diff
@@ -53,18 +58,57 @@ def process_filing_webhook_event(fwe_pk: int) -> FilingWebhookEvent:
     if DO_NOT_POST.search(filing_webhook_event.description):
         return filing_webhook_event
 
+    document = None
     cl_document = lookup_document_by_doc_id(filing_webhook_event.doc_id)
-    document_url = (
-        f"https://storage.courtlistener.com/{cl_document['filepath_local']}"
-        if cl_document["filepath_local"]
-        else None
+    if cl_document["filepath_local"]:
+        document = get_document_from_CL(cl_document["filepath_local"])
+    else:
+        sponsorship = get_active_sponsorship()
+        if sponsorship and filing_webhook_event.pacer_doc_id:
+            purchase_pdf_by_doc_id(filing_webhook_event.doc_id)
+            return filing_webhook_event
+
+    for channel in get_enabled_channels():
+        queue.enqueue(
+            make_post_for_webhook_event,
+            channel.pk,
+            subscription.pk,
+            filing_webhook_event.pk,
+            document,
+            retry=Retry(
+                max=settings.RQ_MAX_NUMBER_OF_RETRIES,
+                interval=settings.RQ_RETRY_INTERVAL,
+            ),
+        )
+
+    return filing_webhook_event
+
+
+@transaction.atomic
+def process_fetch_webhook_event(fwe_pk: int):
+    """Process a RECAP fetch webhook event from CL.
+
+    :param fwe_pk: The PK of the FilingWebhookEvent record.
+    :return: A FilingWebhookEvent object that was updated.
+    """
+    filing_webhook_event = FilingWebhookEvent.objects.get(pk=fwe_pk)
+    subscription = Subscription.objects.get(
+        cl_docket_id=filing_webhook_event.docket_id
     )
 
-    document = None
-    if document_url:
-        document_request = requests.get(document_url, timeout=3)
-        document_request.raise_for_status()
-        document = document_request.content
+    cl_document = lookup_document_by_doc_id(filing_webhook_event.doc_id)
+    document = get_document_from_CL(cl_document["filepath_local"])
+
+    sponsorship = get_active_sponsorship()
+    if sponsorship:
+        Transaction.objects.create(
+            user=sponsorship.user,
+            sponsorship=sponsorship,
+            type=Transaction.DOCUMENT_PURCHASE,
+            amount=3.0
+            if cl_document["page_count"] >= 30
+            else cl_document["page_count"] * 0.10,
+        )
```

Comment:
This feels like it might be well placed into a function. Something like:

```
def log_purchase(...)
```

Also, we have a note field. Let's put something in the note explaining what was bought with docket/document/court info if we have it?

Reply:
You're right about the logic to create a new entry in the transaction table, I will refactor that code into a helper.